src/rstar.jl

```diff
@@ -40,21 +40,22 @@ function rstar(
         throw(ArgumentError("training and test data subsets must not be empty"))
 
     xtable = _astable(x)
+    ycategorical = MLJModelInterface.categorical(ysplit)
 
     # train classifier on training data
-    ycategorical = MLJModelInterface.categorical(ysplit)
-    xtrain = MLJModelInterface.selectrows(xtable, train_ids)
-    fitresult, _ = MLJModelInterface.fit(
-        classifier, verbosity, xtrain, ycategorical[train_ids]
+    xtrain, ytrain = MLJModelInterface.reformat(
+        classifier, MLJModelInterface.selectrows(xtable, train_ids), ycategorical[train_ids]
     )
+    fitresult, _ = MLJModelInterface.fit(classifier, verbosity, xtrain, ytrain)
 
     # compute predictions on test data
-    xtest = MLJModelInterface.selectrows(xtable, test_ids)
+    xtest, ytest = MLJModelInterface.reformat(
+        classifier, MLJModelInterface.selectrows(xtable, test_ids), ycategorical[train_ids]
+    )
```

Comment:
I don't think we can assume that `reformat` will produce any particular output type for `ytest`, so perhaps better to directly index it:
```suggestion
    xtest, = MLJModelInterface.selectrows(classifier, test_ids, xdata)
    ytest = ycategorical[test_ids]
```

Reply:
That would be inconsistent with the suggestion above though? In line with the implementation above we could use
```suggestion
    xtest, ytest = MLJModelInterface.selectrows(classifier, test_ids, xdata, ydata)
```

Reply:
The problem is that `xdata` and `ydata` are now some type that we don't know, because the `reformat` output could be anything. So here `xtest` and `ytest` are correctly a subset of the rows, but we don't know that `ytest` is a vector anymore. e.g. it could be a `NamedTuple` with the vector stored in some field.

Reply:
I assumed the output of `predict` would be of the same type as the reformatted labels. And in that case `_rstar` would fail anyway. But maybe that assumption is not correct.

Reply:
The docs are somewhat ambiguous on this. They [say](https://alan-turing-institute.github.io/MLJ.jl/dev/adding_models_for_general_use) this:

> In the case of Deterministic models, yhat should have the same scitype as the y passed to fit (see above). If y is a CategoricalVector (classification) then elements of the prediction yhat must have a pool == to the pool of the target y presented in training, even if not all levels appear in the training data or prediction itself.

> In the case of Probabilistic models with univariate targets, yhat must be an AbstractVector or table whose elements are distributions. In the common case of a vector (single target), this means one distribution per row of Xnew. 

However, this docs page often makes statements about the inputs `X` and `y` that it clarifies are not applicable if one implements `reformat`. Since there is no `unreformat` method that converts the output of `reformat` to the vector type the user is expected to provide, I think we can surmise that `predict` is expected to return an `AbstractVector` type for the predictions of the same type that the user should provide to `fit`. So passing `predictions` and `ycategorical[test_ids]` to `_rstar` should be fine.

I'm not 100% sure about this, but I don't know what else might make sense. To be 100% sure about this, we would probably need to check what `predict` does for a `Machine` input.

Reply:
This is where `predict` is defined for a `Machine`: https://github.com/JuliaAI/MLJBase.jl/blob/c5d755e157c853850d5f2c4b9693ddf8d9bd469a/src/operations.jl#L130-L139. There's no processing of the output of `predict`, so it must conform to the type expected by a user (vector with the same scitype as the user-provided `y`).

One of the MLJ devs on Slack confirmed that outputs of `predict` are not formatted.